scripts/convert_diffusion_det_to_diffusers.py

```diff
@@ -0,0 +1,126 @@
+      dest_name_prefix = f'encoder.layers.{layer_id}.blocks.{block_id}'
+      _load_weights(f'{src_name_prefix}.norm1.weight', f'{dest_name_prefix}.layernorm_before.weight')
+      _load_weights(f'{src_name_prefix}.norm1.bias', f'{dest_name_prefix}.layernorm_before.bias')
+      _load_weights(f'{src_name_prefix}.attn.relative_position_bias_table', f'{dest_name_prefix}.attention.self.relative_position_bias_table')
+      _load_weights(f'{src_name_prefix}.attn.relative_position_index', f'{dest_name_prefix}.attention.self.relative_position_index')
+
+      # self attention weights.
+      orig_size = orig_model_state[f'{src_name_prefix}.attn.qkv.weight'].shape[0]
+      q_w, k_w, v_w = torch.split(orig_model_state[f'{src_name_prefix}.attn.qkv.weight'], [orig_size//3, orig_size//3, orig_size//3], dim=0)
+      q_b, k_b, v_b = torch.split(orig_model_state[f'{src_name_prefix}.attn.qkv.bias'], [orig_size//3, orig_size//3, orig_size//3], dim=0)
+      state_dict.update({
+          f'{dest_name_prefix}.attention.self.query.weight': q_w,
+          f'{dest_name_prefix}.attention.self.query.bias': q_b,
+
+          f'{dest_name_prefix}.attention.self.key.weight': k_w,
+          f'{dest_name_prefix}.attention.self.key.bias': k_b,
+
+          f'{dest_name_prefix}.attention.self.value.weight': v_w,
+          f'{dest_name_prefix}.attention.self.value.bias': v_b,
+      })
+      
+      _load_weights(f'{src_name_prefix}.attn.proj.weight', f'{dest_name_prefix}.attention.output.dense.weight')
+      _load_weights(f'{src_name_prefix}.attn.proj.bias', f'{dest_name_prefix}.attention.output.dense.bias')
+
+      _load_weights(f'{src_name_prefix}.norm2.weight', f'{dest_name_prefix}.layernorm_after.weight')
+      _load_weights(f'{src_name_prefix}.norm2.bias', f'{dest_name_prefix}.layernorm_after.bias')
+
+      _load_weights(f'{src_name_prefix}.mlp.fc1.weight', f'{dest_name_prefix}.intermediate.dense.weight')
+      _load_weights(f'{src_name_prefix}.mlp.fc1.bias', f'{dest_name_prefix}.intermediate.dense.bias')
+
+      _load_weights(f'{src_name_prefix}.mlp.fc2.weight', f'{dest_name_prefix}.output.dense.weight')
+      _load_weights(f'{src_name_prefix}.mlp.fc2.bias', f'{dest_name_prefix}.output.dense.bias')
+
+    if downsample:
+      _load_weights(f'backbone.bottom_up.layers.{layer_id}.downsample.reduction.weight', f'encoder.layers.{layer_id}.downsample.reduction.weight')
+      _load_weights(f'backbone.bottom_up.layers.{layer_id}.downsample.norm.weight', f'encoder.layers.{layer_id}.downsample.norm.weight')
+      _load_weights(f'backbone.bottom_up.layers.{layer_id}.downsample.norm.bias', f'encoder.layers.{layer_id}.downsample.norm.bias')
+
+  _load_swin_layers(layer_id=0, num_blocks=2)
+  _load_swin_layers(layer_id=1, num_blocks=2)
+  _load_swin_layers(layer_id=2, num_blocks=18)
+  _load_swin_layers(layer_id=3, num_blocks=2, downsample=False)
+
+  # TODO(ishan): check absent layernorm's after layer-{0,1,2}
```

Comment:
@patrickvonplaten @ShoufaChen  can you help shed some light on this?

I am seeing the provided checkpoint has layernorm after every layer as i see in the weights:
``
backbone.bottom_up.norm0.weight torch.Size([128])
backbone.bottom_up.norm0.bias torch.Size([128])
backbone.bottom_up.norm1.weight torch.Size([256])
backbone.bottom_up.norm1.bias torch.Size([256])
backbone.bottom_up.norm2.weight torch.Size([512])
backbone.bottom_up.norm2.bias torch.Size([512])
backbone.bottom_up.norm3.weight torch.Size([1024])
backbone.bottom_up.norm3.bias torch.Size([1024])
```
however, the corresponding model which i loading from HF config corresponding to `microsoft/swin-base-patch4-window7-224-in22k` only has layernorm after the last layer.